### PR TITLE
feat: mask admon_key_id && secret_access_key

### LINKED ISSUE #13860: Feature:  make  access_key_id, secret_access_key don't display  all in show  | desc  connection
**Summary**

```
root@localhost:8000/default> show connections;

SHOW connections

-[ RECORD 1 ]-----------------------------------
          name: myminio
  storage_type: s3
storage_params: access_key_id=minioadmin endpoint_url=http://192.168.1.100:9900 secret_access_key=minioadmin

1 row read in 0.008 sec. Processed 0 row, 0 B (0 row/s, 0 B/s)

root@localhost:8000/default> desc connection myminio;

DESC connection myminio

-[ RECORD 1 ]-----------------------------------
          name: myminio
  storage_type: s3
storage_params: access_key_id=minioadmin endpoint_url=http://192.168.1.100:9900 secret_access_key=minioadmin

1 row read in 0.005 sec. Processed 0 row, 0 B (0 row/s, 0 B/s)

```

### PR DESCRIPTION
**Pull request by guojidan**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #13860  issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13916)

### CODE REVIEW

**Comment by youngsofun** [comment]
we can deal with it in one place:

https://github.com/datafuselabs/databend/blob/main/src/query/ast/src/ast/statements/copy.rs#L305

**Comment by guojidan** [comment]
> we can deal with it in one place:
> 
> https://github.com/datafuselabs/databend/blob/main/src/query/ast/src/ast/statements/copy.rs#L305

`Connection::mask()` look like not related to show/desc connetions, can you explain it more? @youngsofun

**Comment by youngsofun** [comment]
> > we can deal with it in one place:
> > https://github.com/datafuselabs/databend/blob/main/src/query/ast/src/ast/statements/copy.rs#L305
> 
> `Connection::mask()` look like not related to show/desc connetions, can you explain it more? @youngsofun

UserDefinedConnection.storage_params = Connection.conn, they have the same contents.

what about. Connection::new(UserDefinedConnection.storage_params) for the printing of UserDefinedConnection.storage_params?

by the way, Connection::mask(&self) now simply mask everything, maybe we can only mask secret values later.

File: src/query/ast/src/ast/statements/copy.rs [hunk 1]
```diff
@@ -305,7 +305,11 @@ impl Connection {
     pub fn mask(&self) -> Self {
         let mut conns = BTreeMap::new();
         for (k, v) in &self.conns {
-            conns.insert(k.to_string(), mask_string(v, 3));
+            let mut value = v.clone();
+            if k.to_lowercase() == "access_key_id" || k.to_lowercase() == "secret_access_key" {
```
Comment (youngsofun) [inline src/query/ast/src/ast/statements/copy.rs:309]: not only these 2 keys needs masking.
please refer to fn `parse_uri_location`.
if you are not formiliar with this field,  still masking all values is ok.

Comment (guojidan) [reply src/query/ast/src/ast/statements/copy.rs:309]: ok, I will keep masking all values

**Review by youngsofun — APPROVED** [review]

**Pull request merged**